cmd/flightctl-agent/main.go

```diff
@@ -2,74 +2,94 @@ package main
 
 import (
 	"context"
-	"flag"
+	"fmt"
 	"os"
-	"os/signal"
-	"path/filepath"
-	"syscall"
-	"time"
 
 	"github.com/flightctl/flightctl/internal/agent"
-	"github.com/flightctl/flightctl/internal/util"
 	"github.com/flightctl/flightctl/pkg/log"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+	"github.com/spf13/viper"
```

Comment:
At one point, we had discussed whether to remove flags (except one for config file location) and only read from config file. Are Viper et al. giving us a lot of benefit in that scenario? How much weight do we add with them?

Reply:
Good points.

- removing flags feels strange most binaries allow both + ENV? do we have good examples of agents that don't allow flags for configuration? What is the general standard for systemd units to accept  variable params. kubelet and etcd have flags but generally take runtime params via ENV through an ENV file or directly.
- for now I think taking a dep on viper gives us progress in terms of weight we can optimize down the road. IE add the logic directly vs a dep. But is this a priority now?

```
find vendor/github.com/spf13/viper/ -type f -name "*.go" ! -name "*_test.go" -print0 | xargs -0 du -ch
4.0K    vendor/github.com/spf13/viper/file.go
4.0K    vendor/github.com/spf13/viper/file_finder.go
4.0K    vendor/github.com/spf13/viper/flags.go
4.0K    vendor/github.com/spf13/viper/logger.go
8.0K    vendor/github.com/spf13/viper/util.go
64K     vendor/github.com/spf13/viper/viper.go
4.0K    vendor/github.com/spf13/viper/internal/encoding/decoder.go
4.0K    vendor/github.com/spf13/viper/internal/encoding/encoder.go
4.0K    vendor/github.com/spf13/viper/internal/encoding/error.go
4.0K    vendor/github.com/spf13/viper/internal/encoding/dotenv/codec.go
4.0K    vendor/github.com/spf13/viper/internal/encoding/dotenv/map_utils.go
4.0K    vendor/github.com/spf13/viper/internal/encoding/hcl/codec.go
4.0K    vendor/github.com/spf13/viper/internal/encoding/ini/codec.go
4.0K    vendor/github.com/spf13/viper/internal/encoding/ini/map_utils.go
4.0K    vendor/github.com/spf13/viper/internal/encoding/javaproperties/codec.go
4.0K    vendor/github.com/spf13/viper/internal/encoding/javaproperties/map_utils.go
4.0K    vendor/github.com/spf13/viper/internal/encoding/json/codec.go
4.0K    vendor/github.com/spf13/viper/internal/encoding/toml/codec.go
4.0K    vendor/github.com/spf13/viper/internal/encoding/yaml/codec.go
4.0K    vendor/github.com/spf13/viper/internal/features/bind_struct.go
4.0K    vendor/github.com/spf13/viper/internal/features/bind_struct_default.go
148K    total
```

tl;dr 
- we should agree on functionality but perhaps this will bleed until after the demo?
- some binary optimizations will be post MVP. IE seperate module for agent, no dependencies etc

Reply:
It's probably a matter of taste. I see config that's reasonably complex (more than a few args, esp. when config is better structured) and config that may change at runtime (on config file change or on systemd unit reload) being passed in via config file more often. Command line flags seem to be used more for configuring the systemd context in which an app runs (forking policy, pidfiles, log level, config file path) rather than app config itself.

Anyway, was comparing over-the-wire sizes of the binary (after removing debug symbols and compression), and it "only" increased by 9% or 440kB. Quite a bit, but I expected worse TBH. So, yeah, let's go ahead.

Reply:
I think your ask is reasonable if I can do it without the dep quickly I will.